### ISSUE TITLE
Proposal to merge tasks et reminders

### ISSUE THREAD

**Issue by djaiss** [monicahq/monica#821]
The product, as it currently stands, makes it hard to understand the value of tasks compared to reminders.

When you think about it, a reminder is a task with a due date. 

Also, at the moment, you can’t check reminders to indicate that they’ve been processed. Therefore, we don't have a notion of “task is overdue”, which is a really interesting concept in my opinion.

On .com, users barely use tasks. We have less than 800 tasks at the current time, which is nothing compared to the thousands and thousands of reminders people have.

**I propose to merge tasks and reminders.**

This is how I imagine they could work:
* A task is still a task
* A reminder is simply a task with a due date
* Reminders (ie tasks with due dates) can be checked (as they are tasks now)
    * If they are checked before the due date, the reminder won't be sent by email
    * If they are not checked and the due date is reached, the reminder is sent by email
    * The reminder becomes overdue (we can imagine it becomes red in the UI to indicate the urgency)
* In the case of a recurring reminder (like a monthly anniversary)
    * A new task is created right once the reminder is checked 
    * A new task is created as soon as the due date is reached. The current reminder then becomes overdue, and will exist until it is checked.
* We would display the list of upcoming tasks, and the tasks that are overdue, on the dashboard.

**Comment by erdmenchen** [comment]
I dont like the idea of a checkable birthday reminder. I just want to be reminded, no need to mark it as done.

Also I would argue that your argument against the current state of tasks, that it is not an adopted and used feature, does not hold up because the missing part is the ability to mark it as done, like you alredy stated.
This is e.g. also true for birthday presents, I would say. The missing ability to mark a gift idea as "given" makes this feature useless.

Monica is in a critical state: Normaly you make it work and then review and make it pretty. Currently many features are not finished but you are already refactoring them...

Just my two cents. Keep the good work up!

**Comment by samschooler** [comment]
I would vote against merging them completely. They are valuable on their own like @erdmenchen said. However making them more similar, and having code reuse would be a good thing.

My suggestion would be to literally make tasks = reminders (saved in the same table and everything), however they are able to be marked as done. Keep the separate sections, as @erdmenchen said, Birthdays are reminders I want to keep, but not be checkable. Same with club meetings, or a weekly get together. That said I think a valuable addition to tasks would be #827. Adopting this workflow covers tasks and events. For example a task with no due date (how tasks are now) Still are supported, in addition to general time span tasks ("next week" type events), or solidified date tasks ("do this on tuesday").

**Comment by fmcpherson** [comment]
I agree with merging. May I also ask if it is possible for documentation on how to set up Reminders when one does their own installation? It looks like you need to set the configuration for an SMTP server and I need to figure out how to find such a server.
